service-front/module/Application/src/Controller/IndexController.php

```diff
@@ -184,4 +267,90 @@ public function passportNumberAction(): ViewModel
 
         return $view->setTemplate('application/pages/passport_number');
     }
+
+    public function idVerifyQuestionsAction(): ViewModel|Response
+    {
+        $view = new ViewModel();
+        $uuid = $this->params()->fromRoute("uuid");
+        $view->setVariable('uuid', $uuid);
+
+        $form = (new AttributeBuilder())->createForm(IdQuestions::class);
+        $questionsData = $this->opgApiService->getIdCheckQuestions($uuid);
+        $view->setVariable('questions_data', $questionsData);
+        $view->setVariable('question', 'one');
+
+        if (count($this->getRequest()->getPost())) {
+            $formData = $this->getRequest()->getPost();
+
+            $next = $this->getNextQuestion($formData->toArray());
+            if ($next != 'end') {
+                $view->setVariable('question', $next);
+            } else {
+                try {
+                    $check = $this->opgApiService->checkIdCheckAnswers($uuid, ['answers' => $formData->toArray()]);
+
+                    if (! $check) {
+                        return $this->redirect()->toRoute('identity_check_failed', ['uuid' => $uuid]);
+                    }
+
+                    return $this->redirect()->toRoute('identity_check_passed', ['uuid' => $uuid]);
+                } catch (OpgApiException $exception) {
+                    return $this->redirect()->toRoute('identity_check_failed', ['uuid' => $uuid]);
+                }
+            }
+            $form->setData($formData);
+        }
+        $view->setVariable('form', $form);
+
+        return $view->setTemplate('application/pages/identity_check_questions');
+    }
+
+    private function getNextQuestion(array $formdata): string
+    {
+        $question = array_key_last($formdata);
+
+        $sequence = [
+            "one" => "two",
+            "two" => "three",
+            "three" => "four",
+            "four" => "end"
+        ];
+        /**
+         * @psalm-suppress PossiblyNullArgument
+         */
+        return array_key_exists($question, $sequence) ? $sequence[$question] : "";
```

Comment:
Probably easier to use null-coalescing operator:
```suggestion
        return $sequence[$question] ?? "";
```